src/api/callback.cc

```diff
@@ -215,14 +216,14 @@ MaybeLocal<Value> InternalMakeCallback(Environment* env,
 
   Local<Context> context = env->context();
   if (use_async_hooks_trampoline) {
-    MaybeStackBuffer<Local<Value>, 16> args(3 + argc);
+    LocalVector<Value> args(env->isolate(), 3 + argc);
```

Comment:
Did you benchmark this? The point of `MaybeStackBuffer` is to avoid an extra heap allocation, as the name implies, a benefit that gets lost when switching to `LocalVector`

Reply:
I think we probably don't have a choice according to https://chromium-review.googlesource.com/c/v8/v8/+/4905902 - the `Local` handles are not supposed to be heap-allocated in the usual fashion, so a `MaybeStackBuffer<Local<>>` that heap-allocates is invalid in the first place and we were probably just being lucky to dodge some memory failures that could be caused by it. `LocalVector` was added as a V8-acknowledged way to properly heap allocate them.

Reply:
Although, another way to go with this while preserving the stack-allocation part may be to add a different `MaybeStackBuffer` that uses `LocalVector` when it goes heap-allocated, or add some specialization to `MaybeStackBuffer` so that the heap-allocated part uses `LocalVector` (seems messier though).

Reply:
@joyeecheung Yeah, those sound like good ideas for solving this problem to me 🙂

Reply:
Q: How can we disallow heap allocation on Local handles? What's the best way?

Reply:
https://github.com/targos/node/commit/0982497ba47be31ce338efc2950d401e1dc1bbb8

It breaks the world: https://ci.nodejs.org/view/All/job/node-test-commit-arm-debug/15086/nodes=ubuntu2204_debug-arm64/console

Reply:
Let's see if it breaks here as well. Thanks for the tip @targos: https://github.com/cloudflare/workerd/pull/2752